assets/js/modules/analytics-4/utils/data-mock.js

```diff
@@ -0,0 +1,558 @@
+/**
+ * Analytics 4 test data mock.
+ *
+ * Site Kit by Google, Copyright 2023 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * External dependencies
+ */
+import md5 from 'md5';
+import faker from 'faker';
+import invariant from 'invariant';
+import castArray from 'lodash/castArray';
+import { zip, from, Observable } from 'rxjs';
+import { map, reduce, take } from 'rxjs/operators';
+import isPlainObject from 'lodash/isPlainObject';
+
+/**
+ * Internal dependencies
+ */
+import { MODULES_ANALYTICS_4 } from '../datastore/constants';
+import { isValidDateString } from '../../../util';
+import { stringToDate } from '../../../util/date-range/string-to-date';
+
+const ANALYTICS_4_METRIC_TYPES = {
+	totalUsers: 'INTEGER',
+	newUsers: 'INTEGER',
+	sessions: 'INTEGER',
+	conversions: 'INTEGER',
+	screenPageViews: 'INTEGER',
+	engagedSessions: 'INTEGER',
+	averageSessionDuration: 'SECONDS',
+};
+
+const ANALYTICS_4_DIMENSION_OPTIONS = {
+	sessionDefaultChannelGrouping: [
+		'Direct',
+		'Organic Search',
+		'Paid Social',
+		'Organic Social',
+		'Email',
+		'Affiliates',
+		'Referral',
+		'Paid Search',
+		'Video',
+		'Display',
+	],
+	country: [
+		'United States',
+		'United Kingdom',
+		'India',
+		'(not set)',
+		'France',
+		'Ukraine',
+		'Italy',
+		'Mexico',
+	],
+	deviceCategory: [ 'Desktop', 'Tablet', 'Mobile' ],
+	pageTitle: ( i ) => ( i <= 12 ? `Test Post ${ i }` : false ),
+	pagePath: ( i ) => ( i <= 12 ? `/test-post-${ i }/` : false ),
+};
+
+/**
+ * Gets metric key.
+ *
+ * @since n.e.x.t
+ *
+ * @param {string|Object} metric Metric name or object.
+ * @return {string} Metric key.
+ */
+function getMetricKey( metric ) {
+	return metric?.name || metric.toString();
+}
+
+/**
+ * Gets metric type.
+ *
+ * @since n.e.x.t
+ *
+ * @param {string|Object} metric Metric name or object.
+ * @return {string} Type of the metric.
+ */
+function getMetricType( metric ) {
+	return ANALYTICS_4_METRIC_TYPES[ getMetricKey( metric ) ];
+}
+
+/**
+ * Generates and returns metric values.
+ *
+ * @since n.e.x.t
+ *
+ * @param {Array.<Object>} validMetrics Metric list.
+ * @return {Array.<Object>} Array of metric values.
+ */
+function generateMetricValues( validMetrics ) {
+	const values = [];
+
+	validMetrics.forEach( ( validMetric ) => {
+		switch ( getMetricType( validMetric ) ) {
+			case 'INTEGER':
```

Comment:
I think it would be preferable to include the full type here rather than prefixing `TYPE_` further on. That makes it a bit clearer when cross referencing the report output with this code. Also it improves searchability, so this file will appear in the search results when searching for `TYPE_INTEGER` etc.

```suggestion
			case 'TYPE_INTEGER':
```

Obviously, this applies to the other types below.

Reply:
Sounds good. Updated!